### DISCUSSION TITLE
INTERCEPTOR FOR BACK

### DISCUSSION THREAD

**Discussion by kemalinnowise** [serverpod/serverpod#934] [Q&A] [answered]
In my project I have Keycloak connected in FE-side and in the back I need to handle JWT-token passed from FE. So for the moment I don't have any idea how to check wether token is valid on each request. Like in traditional REST In FE side I may set Interceptors for request and response so all this token issues Like store update and etc..  will be handled there and the same I can handle for Back in Java Spring. Any ideas how can I achieve my goal ?

**Comment by vlidholt** [comment, marked as answer]
Hi! It sounds like you will need to build your own extension to the auth module. My best suggestion is to have a look at how the Google or Apple authentication work and use that as a starting point.

https://github.com/serverpod/serverpod/blob/main/modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/google_endpoint.dart
https://github.com/serverpod/serverpod/blob/main/modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/apple_endpoint.dart

We are working on much improved documentation for our authentication, but it may be a few weeks out.
